keanu-project/src/main/java/io/improbable/keanu/vertices/dbltensor/KeanuRandom.java

```diff
@@ -26,4 +33,47 @@ public DoubleTensor nextDouble(int[] shape) {
     public DoubleTensor nextGaussian(int[] shape) {
         return new Nd4jDoubleTensor(nd4jRandom.nextGaussian(shape));
     }
+
+    public DoubleTensor nextGamma(int[] shape, DoubleTensor a, DoubleTensor theta, DoubleTensor k) {
+        List<Double> gammaValues = new ArrayList<>();
+        List<List<Integer>> possibleIndexes = exploreIndexes(shape);
```

Comment:
Very impressive what you've done here. Instead of generating all of the possible indexes and then iterating through them, is there a way to generate the next index as you're iterating?

Reply:
Not too sure what you mean. Do you mean take the sample as we iterate? If so, that's a cracking idea. Will have a go